lib/node_modules/@stdlib/math/base/special/hyp2f1/lib/main.js

```diff
@@ -0,0 +1,284 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var abs = require( '@stdlib/math/base/special/abs' );
+var gamma = require( '@stdlib/math/base/special/gamma' );
+var round = require( '@stdlib/math/base/special/round' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var PINF = require( '@stdlib/constants/math/float64-pinf' );
+var max = require( '@stdlib/math/base/special/max' );
+
+var { hys2f1 } = require( './support.js' );
+var { hyt2f1 } = require( './support.js' );
+
+
+// VARIABLES //
+
+var EPS = 1.0e-13;
+var ETHRESH = 1.0e-12;
+
+
+// MAIN //
+
+/**
+* Evaluates the Gaussian hypergeometric function.
+*
+* @param {number} a - input value
+* @param {number} b - input value
+* @param {number} c - input value
+* @param {number} x - input value
+* @returns {number} function value
+*
+* @example
+* var v = hyp2f1( 1.0, 1.0, 1.0, 0.0 );
+* // returns 1.0
+*/
+function hyp2f1( a, b, c, x ) {
+	var negIntCaOrCb;
+	var negIntA;
+	var negIntB;
+	var err;
+	var aid;
+	var ax;
+	var ia;
+	var ib;
+	var ic;
+	var id;
+	var t1;
+	var d1;
+	var d2;
+	var i;
+	var d;
+	var e;
+	var p;
+	var q;
+	var r;
+	var s;
+	var y;
+
+	negIntA = 0;
+	negIntB = 0;
+	negIntCaOrCb = 0;
+	err = {
+		'v': 0.0
+	};
+	ax = abs( x );
+	s = 1.0 - x;
+	ia = round( a );
+	ib = round( b );
+
+	if ( x === 0.0 ) {
+		return 1.0;
+	}
+
+	d = c - a - b;
+	id = round( d );
+
+	if ( ( a === 0 || b === 0 ) && c !== 0 ) {
+		return 1.0;
+	}
+
+	if ( a <= 0 && abs( a - ia ) < EPS ) { // a is negative integer
+		negIntA = 1;
+	}
+
+	if ( b <= 0 && abs( b - ib ) < EPS ) { // b is negative integer
+		negIntB = 1;
+	}
+
+	if ( d <= -1 && !( abs( d - id ) > EPS ) ) {
+		if ( s < 0 && !( negIntA || negIntB ) ) {
+			return pow( s, d ) * hyp2f1( c-a, c-b, c, x );
+		}
+	}
+
+	if ( d <= 0 && x === 1 && !( negIntA || negIntB )) {
+		// CEPHES: goto hypdiv
+		return PINF;
+	}
+	if ( ax < 1.0 || x === -1.0 ) { // 2F1(a,b;b;x) = (1-x)**(-a)
+		if ( abs( b-c ) < EPS ) { // b = c
+			if ( negIntB ) {
+				y = hyp2f1NegCEqualBC( a, b, x );
+			} else {
+				y = pow( s, -a );
+			}
+			// CEPHES: goto hypdon
+			return y;
+		}
+		if ( abs( a - c ) < EPS ) { // a = c
+			y = pow( s, -b );
+			return y;
+		}
+	}
+	if ( c <= 0.0 ) {
+		ic = round( c );
+		if ( abs( c - ic ) < EPS ) { // c is negative integer
+			// Check termination before explosion
+			if ( negIntA && ( ia > ic ) ) {
+				// CEPHES: goto hypok
+				y = hyt2f1( a, b, c, x, err );
+				return y;
+			}
+			if ( negIntB && ( ib > ic ) ) {
+				// CEPHES: goto hypok
+				y = hyt2f1( a, b, c, x, err );
+				return y;
+			}
+			// CEPHES: goto hypdiv
+			return PINF;
+		}
+	}
+
+	if ( negIntA || negIntB ) { // function in polynomial
+		// CEPHES: goto hypok
+		y = hyt2f1( a, b, c, x, err );
+		return y;
+	}
+	t1 = abs( b - a );
+	if ( x < -2.0 && abs( t1 - round(t1) ) > EPS ) {
+		/* This transform has a pole for b-a integer, and
```

Comment:
This can be converted to a single line comment; unless formatting is necessary, e.g., to show an equation, project convention is to use unwrapped single line comments `//`.

Reply:
Ok, you got it.